### DISCUSSION TITLE
Is it possible to implement a widget to display makie's plotting?

### DISCUSSION THREAD

**Discussion by FaresX** [Gnimuc/CImGui.jl#112] [Ideas]
I have used ImPlot.jl to render plots. But it is not satisfying and the fps go down steeply when I render a heatmap plot with a lot of data. I roughly scanned the source codes of Gtk4Makie.jl and found that it is possible to implement a **Screen** to build a new backend for Makie and render more than one plot in a single window with all the events, but I am not familiar with this field and would like to ask if you everyone have any ideas. Thank you!

**Comment by JamesWrigley** [comment]
I also would like to do this but haven't gotten around to it, see: https://discord.com/channels/996787732149981214/1214594085969596436

Out of curiosity, what was the performance like with the heatmaps?

**Reply by FaresX** [reply]
When data points exceed approximately 100000, fps will decrease from 60 to 30. Multiple plots can be worse. By the way, I cannot see anything through this link.

**Reply by JamesWrigley** [reply]
Oh, yeah you'll probably have to join their Discord first: https://discord.gg/dwxUVEH6Tz

If you're working with large line plots in particular, then you could apply the LTTB algorithm before plotting: https://discourse.julialang.org/t/how-to-plot-10-billion-datapoint-time-series-efficiently/81228/12

**Reply by FaresX** [reply]
I just had a look at the post and found you thought about it a month ago. Do you have any ideas now? I think I should start by learning OpenGL, which is very unfamiliar to me. Additionally, downsampling is not a good practice compared to turing to Makie because I want to see the real-time raw data.

**Reply by JamesWrigley** [reply]
No I haven't thought about it much since then, but I'd start with the suggestions in that thread. Downsampling with LTTB can definitely be good practice :) I also use it for real-time raw data and it works very well, you'll never be able to distinguish a 100,000 segment line plot on a normal display anyway. But ideally the downsampling would be done dynamically by the plotting library so that e.g. when you zoom in a small region of the plot it won't downsample and you'll see any high frequency data.

**Reply by FaresX** [reply]
All right, I'll try that later. In addition, if I make some progress on ImGuiMakie, I will come back here or look forward to your good news.

**Comment by JamesWrigley** [comment]
Closing this because we have some experimental support for GLMakie now :octopus: https://gnimuc.github.io/CImGui.jl/stable/makie/
